api/src/main/java/net/kyori/adventure/audience/Audience.java

```diff
@@ -24,163 +24,199 @@
+ * new Minecraft media such as the player list, its interface will remain stateless
+ * and any new methods will be stubbed by default.</p>
+ *
+ * @see ForwardingAudience
+ * @since 1.0
+ * @version 1.0
  */
 public interface Audience {
   /**
    * Gets an audience that does nothing.
    *
-   * @return an audience
+   * @since 1.0
+   * @return a do-nothing audience
    */
   static @NonNull Audience empty() {
     return EmptyAudience.INSTANCE;
   }
 
   /**
-   * Creates an audience that delegates to an array of audiences.
+   * Creates an audience that forwards to many other audiences.
    *
-   * @param audiences the delegate audiences
+   * @since 1.0
+   * @see ForwardingAudience
+   * @param audiences an array of audiences, can be empty
    * @return an audience
    */
   static @NonNull Audience of(final @NonNull Audience@NonNull... audiences) {
-    final int length = audiences.length;
-    if(length == 0) {
-      return empty();
-    } else if(length == 1) {
-      return audiences[0];
+    switch(audiences.length) {
+      case 0: return empty();
+      case 1: return audiences[0];
+      default: return of(Arrays.asList(audiences));
     }
-    return of(Arrays.asList(audiences));
   }
 
   /**
-   * Creates an audience that delegates to a collection of audiences.
+   * Creates an audience that forwards to many other audiences.
    *
-   * @param audiences the delegate audiences
+   * <p>The underlying <tt>Iterable</tt> is not copied, therefore any changes
+   * made will be reflected in <tt>Audience</tt>.</p>
+   *
+   * @since 1.0
+   * @see ForwardingAudience
+   * @param audiences an iterable of audiences, can be empty
    * @return an audience
    */
   static @NonNull Audience of(final @NonNull Iterable<? extends Audience> audiences) {
-    return (MultiAudience) () -> audiences;
+    return (ForwardingAudience) audiences;
   }
 
   /**
-   * Creates an audience that weakly delegates to another audience.
+   * Sends a chat message.
    *
-   * @param audience the delegate audience
-   * @return an audience
-   */
-  static @NonNull Audience weakOf(final @Nullable Audience audience) {
-    return audience instanceof WeakAudience || audience instanceof EmptyAudience ? audience : new WeakAudience(audience);
-  }
-
-  // ------------------
-  // ---- Messages ----
-  // ------------------
-
-  /**
-   * Sends a message.
-   *
-   * @param message the message
+   * @since 1.0
+   * @see Component
+   * @param message a message
    */
   void sendMessage(final @NonNull Component message);
 
-  // --------------------
-  // ---- Action Bar ----
-  // --------------------
-
   /**
    * Sends a message on the action bar.
    *
-   * @param message the message
+   * @since 1.0
+   * @see Component
+   * @param message a message
    */
-  void sendActionBar(final @NonNull Component message);
-
-  // ----------------
-  // ---- Titles ----
-  // ----------------
+  default void sendActionBar(final @NonNull Component message) {}
```

Comment:
Why becoming `default`?

Reply:
The contract is `sendMessage` is required, but everything else is stub by default. This important since lots of implementations will be extending `Audience`. We don't want push back that we're going to break everyone because we add some new method.

Reply:
Why make `sendMessage` required and everything else not?

Reply:
If an `Audience` doesn't implement the basic message, it's probably not the right abstraction? We could also make this not require too.

Reply:
We already go against a required `sendMessage` in other parts of our API -- everywhere we return Audience.empty() instead of null, for example. It would only create confusion to claim that sendMessage is always required.

I'm also against Audience itself having default implementations -- maybe a subinterface or abstract audience could have the defaults?

Reply:
Defaulting on the man interface is the easiest way to future proof it, while also allowing implementors to extend Audience directly and not though a stub. Given the flexibility, I can also see plugins implementing this in their own `MyPluginPlayer` interfaces and what not. It will be a lot easier for us and them when we add the new `playX` feature, it doesn't break any code.